pkg/encoding/v1/encoding_test.go

```diff
@@ -144,9 +144,77 @@ func TestEnvVariable_UnmarshalYAML(t *testing.T) {
 	}
 }
 
+func TestService_UnmarshalYAML(t *testing.T) {
+
+	var positiveReplica int32 = 3
+
+	tests := []struct {
+		Name           string
+		ExpectedSucess bool
+		RawService     string
+		Service        *Service
+	}{
+		{
+			"Giving replica as int to see if it is parsed",
+			true, `
+name: frontend
+replicas: 3
+containers:
+- image: tomaskral/kompose-demo-frontend:test
+`,
+			&Service{
+				Name:     "frontend",
+				Replicas: &positiveReplica,
+				Containers: []Container{
+					{
+						Image: "tomaskral/kompose-demo-frontend:test",
+					},
+				},
+			},
+		},
+
+		{
+			"Giving replica as string to see parsing fails",
+			false, `
+name: frontend
+replicas: notint
+containers:
+- image: tomaskral/kompose-demo-frontend:test
+`,
+			nil,
+		},
+	}
+
+	for _, test := range tests {
+		t.Logf("Running test: %q", test.Name)
```

Comment:
please use t.Run where you can pass the name

we are moving to using subtests (https://blog.golang.org/subtests)

https://github.com/tnozicka/opencompose/blob/a21430158830bff240a8252aa4b609e994907238/pkg/encoding/v1/encoding_test.go#L207